lib/Doctrine/ORM/Query/Lexer.php

```diff
@@ -126,10 +128,11 @@ public function __construct($input)
     protected function getCatchablePatterns()
     {
         return array(
-            '[a-z_\\\][a-z0-9_\:\\\]*[a-z0-9_]{1}',
-            '(?:[0-9]+(?:[\.][0-9]+)*)(?:e[+-]?[0-9]+)?',
-            "'(?:[^']|'')*'",
-            '\?[0-9]*|:[a-z_][a-z0-9_]*'
+            '[a-z_][a-z0-9_]*\:[a-z_][a-z0-9_]*', // aliased name
```

Comment:
I don't understand why you're separating them. You're increasing the complexity of the regex (which is already not so simple) for - what I can see so far, top-to-bottom reviewing - no reason.